hyperspy/io_plugins/emd.py

```diff
@@ -0,0 +1,402 @@
+        # Add comments:
+        self.comments = comments
+        # Make sure the signals are added properly to signals:
+        self.signals = {}
+        for name, signal in signals.iteritems():
+            self.add_signal(signal, name)
+
+    def __getitem__(self, key):
+        # This is for accessing the raw data easily. For the signals use emd.signals[key]!
+        return self.signals[key].data
+
+    def _write_signal_to_group(self, signal_group, signal):
+        self._log.debug('Calling _write_signal_to_group')
+        # Save data:
+        dataset = signal_group.require_group(signal.metadata.General.title)
+        maxshape = tuple(None for _ in signal.data.shape)
+        dataset.create_dataset('data', data=signal.data, chunks=True, maxshape=maxshape)
+        # Iterate over all dimensions:
+        for i in range(len(signal.data.shape)):
+            key = 'dim{}'.format(i+1)
+            axis = signal.axes_manager._axes[i]
+            offset = axis.offset
+            scale = axis.scale
+            dim = dataset.create_dataset(key, data=[offset, offset+scale])
+            name = axis.name
+            from traits.trait_base import _Undefined
+            if type(name) is _Undefined:
+                name = ''
+            dim.attrs['name'] = name
+            units = axis.units
+            if type(units) is _Undefined:
+                units = ''
+            else:
+                units = '[{}]'.format('_'.join(list(units)))
+            dim.attrs['units'] = units
+        # Write metadata:
+        dataset.attrs['emd_group_type'] = 1
+        for key, value in signal.metadata.Signal:
+            try:  # If something h5py can't handle is saved in the metadata:
+                dataset.attrs[key] = value
+            except Exception:
+                pass  # just don't add what can't be added!
+
+    def _read_signal_from_group(self, name, group, load_to_memory=True):
+        self._log.debug('Calling _read_signal_from_group')
+        from hyperspy import signals
+        # Extract essential data:
+        data = group.get('data')
+        if load_to_memory:
+            data = np.asanyarray(data)
+        record_by = group.attrs.get('record_by', '')
+        # Create Signal, Image or Spectrum:
+        if record_by == 'spectrum':
+            signal = signals.Spectrum(data)
+        if record_by == 'image':
+            signal = signals.Image(data)
+        else:
+            signal = signals.Signal(data)
+        # Set signal properties:
+        signal.set_signal_origin = group.attrs.get('signal_origin', '')
+        signal.set_signal_type = group.attrs.get('signal_type', '')
+        # Iterate over all dimensions:
+        for i in range(len(data.shape)):
+            dim = group.get('dim{}'.format(i+1))
+            signal.axes_manager[i].name = dim.attrs.get('name', '')
+            units = re.findall('[^_\W]+', dim.attrs.get('units', ''))
+            signal.axes_manager[i].units = ''.join(units)

```

Comment:
...Actually you could add a test that checks this when saving/loading (that axes do not get mixed)